app/services/oidc.ts

```diff
@@ -0,0 +1,165 @@
+import Service, { inject as service } from '@ember/service';
+import RouterService from '@ember/routing/router-service';
+import { task } from 'ember-concurrency';
+import IntlService from 'ember-intl/services/intl';
+
+import NetworkService from 'irene/services/network';
+
+interface OidcResponse {
+  valid: boolean;
+  redirect_url: null | string;
+  error: null | {
+    code: string;
+    description: string;
+  };
+}
+
+type ValidateOidcTokenResponse = OidcResponse;
+
+type OidcAuthorizeResult = OidcResponse;
```

Comment:
Are the `ValidateOidcTokenResponse`, `OidcValidationResult`, and `OidcAuthorizeResult` types required?